AudioDeviceExample/AudioDevices/ExampleCoreAudioDevice.m

```diff
@@ -454,7 +454,10 @@ - (void)handleValidRouteChange {
         NSLog(@"The rendering format changed. Restarting with %@", activeFormat);
         // Signal a change by clearing our cached format, and allowing TVIAudioDevice to drive the process.
         _renderingFormat = nil;
-        TVIAudioDeviceFormatChanged(self.renderingContext->deviceContext);
+
+        @synchronized(self) {
+            TVIAudioDeviceFormatChanged(self.renderingContext->deviceContext);
+        }
```

Comment:
Do we need to check if `self.renderingContext` is still valid at this point? It might have changed since we enqueued the original message.